### PR TITLE
ctl: support using task-file when need task-name (#854)

### PR DESCRIPTION
**Pull request by lance6716**
manually cherry-pick #854 

---

### What problem does this PR solve? 
part of #826

### What is changed and how it works?
support using task-file when need task-name

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

File: dm/ctl/common/util.go [hunk 1]
```diff
@@ -210,3 +211,22 @@ func IsDDL(sql string) (bool, error) {
 		return false, nil
 	}
 }
+
+// GetTaskNameFromArgOrFile tries to retrieve name field from the file if arg is filename-like, otherwise returns arg directly
+func GetTaskNameFromArgOrFile(arg string) string {
+	if !strings.HasSuffix(arg, ".yaml") {
```
Comment (csuzhangxc) [inline dm/ctl/common/util.go:217]: some people may name a YAML file with the suffix of `yml` rather than `yaml`. 😂

**Comment by lance6716** [comment]
> LGTM. do we also need to update the doc of v1.0, right?

https://github.com/pingcap/docs-dm/pull/220 will cherry-pick to 1.0

File: dm/ctl/common/util.go (hunk 1 above)
Comment (lance6716) [reply dm/ctl/common/util.go]: done in https://github.com/pingcap/dm/pull/892/commits/6d1aede47719764e2eac05009255833c87f349c2

Comment (csuzhangxc) [reply dm/ctl/common/util.go]: we also need to update the master branch now.

**Pull request merged**